concourse/setup.py

```diff
@@ -552,6 +584,56 @@ def deploy_or_upgrade_concourse(
         subprocess.run(subprocess_args, check=True, cwd=temp_dir, env=helm_env)
 
 
+def deploy_or_upgrade_webhook_dispatcher(
+        webhook_dispatcher_cfg: WebhookDispatcher,
+        chart_dir: str,
+        cfg_factory: ConfigFactory,
+        kubernetes_cfg: KubernetesConfig,
+        deployment_name: str='webhook-dispatcher',
+):
+    """Deploys (or upgrades) webhook dispatcher via Helm CLI"""
+    helm_executable = ensure_helm_setup()
+    chart_dir = util.existing_dir(chart_dir)
+
+    namespace = deployment_name
+
+    # create namespace if absent
+    namespace_helper = kube_ctx.namespace_helper()
+    if not namespace_helper.get_namespace(namespace):
+        namespace_helper.create_namespace(namespace)
+
+    WEBHOOK_DISPATCHER_HELM_VALUES_FILE_NAME = 'webhook_dispatcher_helm_values'
+    KUBECONFIG_FILE_NAME = 'kubecfg'
+
+    # prepare subprocess args using relative file paths for the values files
+    subprocess_args = [
+        helm_executable, "upgrade", "--install",
+        "--recreate-pods",
+        "--wait",
+        "--namespace", namespace,
+        # Use Helm's value-rendering mechanism to merge value-sources.
+        "--values", WEBHOOK_DISPATCHER_HELM_VALUES_FILE_NAME,
+        namespace, # release name is the same as namespace name
+        chart_dir,
+    ]
+
+    helm_env = os.environ.copy()
+    # set KUBECONFIG env-var in the copy to relative file path
+    helm_env['KUBECONFIG'] = KUBECONFIG_FILE_NAME
```

Comment:
we now have multiple places in this very module in which a helm subprocess is called with a patched environment to pass kubeconfig. do not continue code duplication. instead, please factor out a function to call a (helm/kubectl) subprocecess with patched env. pbly should be done as a ctx-mgr